lib/ruboty/slack_rtm/client.rb

```diff
@@ -65,7 +65,13 @@ def keep_connection
         Thread.start do
           loop do
             sleep(30)
-            @client.send('', type: 'ping')
+            begin
+              @client.send('', type: 'ping')
+            rescue => e
+              Ruboty.logger.error("#{e.class}: #{e.message}\n#{e.backtrace.join("\n")}")
+              @queue.enq(CONNECTION_CLOSED)
+              break
+            end
```

Comment:
Handle exception in `Thread` to escape from `#main_loop`.

https://github.com/rosylilly/ruboty-slack_rtm/blob/040541fddbc0e98128f8cce318af6abc7951d128/lib/ruboty/slack_rtm/client.rb#L35-L45
https://github.com/rosylilly/ruboty-slack_rtm/blob/040541fddbc0e98128f8cce318af6abc7951d128/lib/ruboty/adapters/slack_rtm.rb#L98-L105